### ISSUE TITLE
Clean up license notices

### ISSUE THREAD

**Issue by Miles-Garnsey** [k8ssandra/k8ssandra#1164]
**Is your feature request related to a problem? Please describe.**

This is 100% a nit that I am picking, but we have different license text at the top of every file in every repo under k8ssandra. It is unsightly.

**Describe the solution you'd like**
Can we write a script to go through and either remove them all, or put something in CI to add license text to all files on commit?

**Comment by adutra** [comment]
In go-cassandra-native-protocol we've been using https://github.com/google/addlicense. Works like a charm.

**Comment by bradfordcp** [comment]
Agreed that this is a pretty big nit. Isn't it required by the Apache License to be at the top of every file?

**Comment by adutra** [comment]
IMO this should be high priority.

**Comment by bradfordcp** [comment]
Agreed, I was going to ask for an estimate here, but assume we also need a ticket for each of the other repos.

Would it be better to think about this as an epic or just pertaining to this repo with other tickets needing to be created and estimated?

**Comment by bradfordcp** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/K8ssandra-Grooming-623b197f1fcd4800110540ef/issues/k8ssandra/k8ssandra/1164?planning-poker) with ZenHub @jsanda @Miles-Garnsey @adejanovski @jeffbanks

**Comment by Miles-Garnsey** [comment]
Do we need to just write a script that detects the presence of the Apache license and adds it if not present? That's a 2 day Python job IMO (being generous).

Add another three days to get this automated within the makefile AND checked within GHA actions, so that commits don't pass without licenses added and licenses get added automatically as part of the `make build` step.

**Comment by Miles-Garnsey** [comment]
Even better, so we can skip the initial 2 day implementation as we don't need to implement something new. We should still look to build the addlicense step into GHA and the make steps though IMO, to avoid excess noise in the commits. 

I checked and `addlicense` is idempotent (i.e. won't add license text if it is already there) so I think it will suit our needs.

Call it 2-3 days to 
1. Wrangle the workflow definitions in GHA and ensure that addlicense has been run on all files.
2. Add `addlicense` as a dependancy to `make build` so that it is run before folks commit. (Maybe we need a `make pre-commit` step to run all of our linting tasks?)

**Comment by jeffbanks** [comment]
@burmanm since we aren't required to include at the top of every file, should I simply verify in our repositories a single LICENSE.txt file for Apache?  Based on your comment here, do we really need to do anything beyond this manual check?

**Comment by burmanm** [comment]
> @burmanm since we aren't required to include at the top of every file, should I simply verify in our repositories a single LICENSE.txt file for Apache?  Based on your comment here, do we really need to do anything beyond this manual check?

If some of our files have the license text, then we probably need the consistency. So remove from others or add to all. I would ask legal's opinion what they want, although the license itself to me is clear that only a single file is needed.

**Comment by jeffbanks** [comment]
Submitted question to legal for clarification.

**Comment by jeffbanks** [comment]
Identified from legal through correspondence in the following ticket, we will need to ensure LICENSE and NOTICE files are included in our K8ssandra repositories.  Only requires the optional header license information when the source files are expected to be detached from the repositories for other usage.

Ticket: https://datastaxlegal.zendesk.com/hc/en-us/requests/8866

**Comment by jeffbanks** [comment]
Additional information specific to the NOTICE file included here:
https://opensource.stackexchange.com/questions/7872/how-to-properly-assemble-notice-file-for-new-software-under-apache-license-2-0

No need to include the NOTICE at this time, as this is not an Apache project.  The LICENSE inclusion will suffice.

**Comment by jeffbanks** [comment]
The only repository for K8ssandra projects not having a LICENSE file in the repository was the recently added `k8ssandra-testutils` repository.  

An [issue](https://github.com/k8ssandra/k8ssandra-testutils/issues/4) was opened to track that fix.
